### PR TITLE
chore: migrate common dict int tests

### PR DESCRIPTION
**Pull request by pgautier404**
This commit migrates dictionary integration tests to the shared integration tests package for reuse between the nodejs and web SDKs, adding implementations for each operation to the web SDK data client. It also removes methods from the nodejs CacheClient that are now defined in the AbstractCacheClient, fixes encoding of collection names in list operations, and adds a test to verify deleting list collections from the cache.

### CODE REVIEW

File: scripts/build-and-test-package.sh [hunk 1]
```diff
@@ -11,6 +11,5 @@ echo "building and testing package: ${1}"
 ${ROOT_DIR}/scripts/build-package.sh ${1}
 
 pushd ${ROOT_DIR}/packages/${1}
-    npm run unit-test
-    npm run integration-test
+    npx jest --maxWorkers 1
```
Comment (cprice404) [inline scripts/build-and-test-package.sh:14]: This was for me for debugging, I don't think we actually want to commit it.

**Review by cprice404 — COMMENTED** [review]
Just one minor nit

File: scripts/build-and-test-package.sh (hunk 1 above)
Comment (pgautier404) [reply scripts/build-and-test-package.sh:14]: Ok, I'll revert it. It did, however, fix a persistent local problem I've been having with tests failing with `Rate exceeded` errors.

Comment (cprice404) [reply scripts/build-and-test-package.sh]: Yeah it is useful for development.  Feel free to add another bash script that uses that flag, or another command in the package.json scripts section.  But that file that you had changed gets used by CI.

**Review by cprice404 — COMMENTED** [review]
I'm good with this now but I would like to hold off on merging it until Matt has resolved the issue he's having with trying to use the web sdk from npm.js, because there is a user on discord waiting for that.

File: packages/client-sdk-web/package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
   "scripts": {
     "prebuild": "eslint . --ext .ts",
     "test": "jest",
-    "unit-test": "jest unit",
+    "unit-test": "jest unit --passWithNoTests",
```
Comment (cprice404) [inline packages/client-sdk-web/package.json:18]: why is this necessary?  If there is a file that doesn't have a test in it can we just delete that file?

Comment (pgautier404) [reply packages/client-sdk-web/package.json]: Hm, actually, I think I need to put it back because the `build-and-test-package` script is explicitly calling it. This allows that script to call `unit-test` and allows `unit-test` to no-op.

Comment (pgautier404) [reply packages/client-sdk-web/package.json:18]: Added back in 63af729 to support `build-and-test-package` script.

Comment (cprice404) [reply packages/client-sdk-web/package.json:18]: I don't like the passWithNoTests flag.  Do we not have any unit tests?  Could we just add a "hello world" one that asserts true == true?

I have seen issues in the past where someone misconfigured a test source dir and then a flag like this would cause the build to pass even if there were tons of failing unit tests because they didn't actually end up getting run.

Comment (cprice404) [reply packages/client-sdk-web/package.json:18]: to be clear - i want to keep the unit tests directive but i want to get rid of that flag :)

**Review by cprice404 — COMMENTED** [review]
ready to go except I'm gonna be annoying about that passWithNoTests flag :)

File: packages/client-sdk-web/package.json (hunk 2 above)
Comment (pgautier404) [reply packages/client-sdk-web/package.json]: Yeah, that's fair. I added a no-op unit test in b579c3d.

**Review by cprice404 — APPROVED** [review]
hooray
